tests/MoBi.Tests/Core/Commands/RenameContainerCommandSpecs.cs

```diff
@@ -0,0 +1,107 @@
+         _container = new Container().WithName("A").WithParentContainer(_parent);
+         _child1 = new Container().WithName("A").WithParentContainer(_container);
+         _child2 = new Container().WithName("B").WithParentContainer(_container);
+         _child3 = new Container().WithName("C").WithParentContainer(_container);
+         _otherContainer = new Container().WithName("B").WithParentContainer(_parent);
+         _otherChild1 = new Container().WithName("A").WithParentContainer(_otherContainer);
+         _otherChild2 = new Container().WithName("B").WithParentContainer(_otherContainer);
+
+         _objectPathFactory = new ObjectPathFactoryForSpecs();
+         _spatialStructure = new SpatialStructure
+         {
+            NeighborhoodsContainer = new Container().WithName(Constants.NEIGHBORHOODS)
+         };
+         _neighborhood1 = new NeighborhoodBuilder
+         {
+            FirstNeighborPath = _objectPathFactory.CreateAbsoluteObjectPath(_child1),
+            SecondNeighborPath = _objectPathFactory.CreateAbsoluteObjectPath(_child2),
+            Name = "N1"
+         };
+
+         _neighborhood2 = new NeighborhoodBuilder
+         {
+            FirstNeighborPath = _objectPathFactory.CreateAbsoluteObjectPath(_otherChild1),
+            SecondNeighborPath = _objectPathFactory.CreateAbsoluteObjectPath(_otherChild2),
+            Name = "N2"
+         };
+
+         _spatialStructure.AddNeighborhood(_neighborhood1);
+         _spatialStructure.AddNeighborhood(_neighborhood2);
+         _spatialStructure.AddTopContainer(_parent);
+
+         _context = A.Fake<IMoBiContext>();
+         A.CallTo(() => _context.ObjectPathFactory).Returns(new ObjectPathFactoryForSpecs());
+      }
+   }
+
+   public class When_renaming_a_container_that_is_not_used_in_any_neighborhood : concern_for_RenameContainerCommand
+   {
+      protected override void Because()
+      {
+         sut = new RenameContainerCommand(_child3, "NEW_NAME", _spatialStructure).Run(_context);
+      }
+
+      [Observation]
+      public void should_not_rename_the_path_in_the_neighborhoods()
+      {
+         _neighborhood1.FirstNeighborPath.PathAsString.ShouldBeEqualTo("A|A|A");
+         _neighborhood1.SecondNeighborPath.PathAsString.ShouldBeEqualTo("A|A|B");
+
+         _neighborhood2.FirstNeighborPath.PathAsString.ShouldBeEqualTo("A|B|A");
+         _neighborhood2.SecondNeighborPath.PathAsString.ShouldBeEqualTo("A|B|B");
+      }
+   }
+
+   public class When_renaming_a_container_that_is_used_in_multiple_neighborhood : concern_for_RenameContainerCommand
+   {
+      protected override void Because()
+      {
+         sut = new RenameContainerCommand(_container, "NEW_NAME", _spatialStructure).Run(_context);
+      }
+
+      [Observation]
+      public void should_rename_the_path_in_the_neighborhoods_not_referencing_this_container()
+      {
+         _neighborhood1.FirstNeighborPath.PathAsString.ShouldBeEqualTo("A|NEW_NAME|A");
```

Comment:
Funny I wrote this test and did not check the run but it fails. And of course, my implementation is wrong. I just cannot check for the exact path but somehow it should just be subpart lol. I'll fix it tomorrow